### PR TITLE
User guide: document basic provisioning and deprovisioning

### PR DESCRIPTION
**Pull request by dtantsur**
We have some information in the quick start guide, but judging by the
questions we receive recently, it is not enough.

Note that I'm putting the document before all other basic features
because it's something that 100% users need to read.

Signed-off-by: Dmitry Tantsur <dtantsur@protonmail.com>

### CODE REVIEW

File: docs/user-guide/src/bmo/provisioning.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Provisioning and Deprovisioning
+
+The most fundamental feature of Metal3 BareMetal Operator is provisioning of
```
Comment (tuminoid) [inline docs/user-guide/src/bmo/provisioning.md:3]:
```suggestion
The most fundamental feature of Metal3 Bare Metal Operator is provisioning of
```
Once again, we should standardize the spelling of our components. RIght now, the spelling used by you is found 6 times, while the suggested spelling is found 20+ times (and `Baremetal Operator` is found another 20+ times :roll_eyes: )

File: docs/user-guide/src/bmo/provisioning.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+## Deprovisioning
+
+To remove an instance from the host and make it available for new deployments,
+remove the `image`, `userData`, `networkData`, `metaData` and `customDeploy`
+fields (if present). Depending on the host configuration, it will either start
+the [automated cleaning](./automated_cleaning.md) process or will become
+`available` right away.
+
+## Re-provisioning
```
Comment (tuminoid) [inline docs/user-guide/src/bmo/provisioning.md:113]:
```suggestion
## Reprovisioning
```
Same with this spelling. Both are legit, but without dash is prevalent and also seems to be more common of the spellings. This occurs just few times, while reprovision occurs much more often.

File: docs/user-guide/src/bmo/provisioning.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+If you want to apply a new image or new user or network data to the host, you
+need to deprovision it and provision again. This can be done in two ways:
+
+- If the URL of the image changes, the re-provisioning process will start
+  automatically. Make sure to update the user and network data in
+  the same edit operation.
```
Comment (tuminoid) [inline docs/user-guide/src/bmo/provisioning.md:120]: Same or prior edit, maybe? The warning couple lines below suggest you can edit user data and network data without triggering reprovisioning.

**Review by tuminoid — COMMENTED** [review]
Great addition!

Left some comments on the spelling, feel free to ignore if not agreeing.

**Review by tuminoid — COMMENTED** [review]
/lgtm

Thanks for adding another piece of documentation!

File: docs/user-guide/src/bmo/provisioning.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+A freshly enrolled host gets provisioned when the two conditions are met:
+
+- the state is `available` (see [state machine](./state_machine.md)),
+- either its `image` field or its `customDeploy` field is not empty.
+
+**NOTE:** `customDeploy` is an advanced feature that is not covered in this
+document.
+
+Two start the provisioning process, you need at least two bits of information:
```
Comment (elfosardo) [inline docs/user-guide/src/bmo/provisioning.md:18]: nit: s/Two/To

File: docs/user-guide/src/bmo/provisioning.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+To remove an instance from the host and make it available for new deployments,
+remove the `image`, `userData`, `networkData`, `metaData` and `customDeploy`
+fields (if present). Depending on the host configuration, it will either start
+the [automated cleaning](./automated_cleaning.md) process or will become
+`available` right away.
+
+## Reprovisioning
+
+If you want to apply a new image or new user or network data to the host, you
+need to deprovision it and provision again. This can be done in two ways:
```
Comment (elfosardo) [inline docs/user-guide/src/bmo/provisioning.md:116]: nit: deprovision and provision it again

**Review by elfosardo — COMMENTED** [review]
looks good!
if you want to fix the nits I'll hold the approval, otherwise it's ok to go

**Comment by elfosardo** [comment]
/lgtm
lol I thought I was in the approvers list!

**Review by tuminoid — APPROVED** [review]

**Review by kashifest — COMMENTED** [review]
/approve

**Pull request merged**